generic/OpenMP_Support.F90

```diff
@@ -570,6 +576,67 @@ subroutine set_services(gc, rc)
             end do
            _RETURN(ESMF_SUCCESS)
         end subroutine set_services
+
     end function make_subgridcomps
 
+    subroutine copy_callbacks(state, multi_states, rc)
+       use mapl_ESMF_Interfaces
+       use mapl_CallbackMap
+       type(ESMF_State), intent(inout) :: state
+       type(ESMF_State), intent(inout) :: multi_states(:)
+       integer, optional, intent(out) :: rc
+
+       integer :: n_multi, i
+       integer :: status
+       type(CallbackMethodWrapper), pointer :: wrapper
+       type(CallbackMap), pointer :: callbacks
+       type(CallbackMapIterator) :: iter
+
+       n_multi = size(multi_states)
+       call get_callbacks(state, callbacks, _RC)
+       if (associated(callbacks)) then
+          associate (e => callbacks%end())
+            iter = callbacks%begin()
+            do
+               wrapper => iter%second()
+               do i = 1, n_multi
+                  call ESMF_MethodAdd(multi_states(i), iter%first(), wrapper%userRoutine, _RC)
+               end do
+               call iter%next()
+            end do
+          end associate
+       end if
+
+       _RETURN(ESMF_SUCCESS)
+
+    end subroutine copy_callbacks
+
+    subroutine get_callbacks(state, callbacks, rc)
```

Comment:
`get_callbacks()` should maybe me moved to some other location.   MAPL_Generic ought not to depend so explicitly on this layer.   But if no obvious choice presents itself, it can stay here for now.